okhttp/src/test/java/io/grpc/okhttp/OkHttpProtocolNegotiatorTest.java

```diff
@@ -0,0 +1,159 @@
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.grpc.okhttp;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.common.collect.ImmutableList;
+
+import io.grpc.okhttp.internal.Platform;
+import io.grpc.okhttp.internal.Protocol;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mockito;
+
+import java.io.IOException;
+
+import javax.net.ssl.SSLSocket;
+
+/**
+ * Tests for {@link OkHttpProtocolNegotiator}.
+ */
+@RunWith(JUnit4.class)
+public class OkHttpProtocolNegotiatorTest {
+  @Rule public final ExpectedException thrown = ExpectedException.none();
+
+  private SSLSocket sock = mock(SSLSocket.class);
+
+  @Test
+  public void createNegotiator_isAndroid() {
+    ClassLoader cl = new ClassLoader(this.getClass().getClassLoader()) {
+      @Override
+      protected Class<?> findClass(String name) throws ClassNotFoundException {
+        // Just don't throw.
+        if ("com.android.org.conscrypt.OpenSSLSocketImpl".equals(name)) {
+          return null;
+        }
+        return super.findClass(name);
+      }
+    };
+
+    OkHttpProtocolNegotiator negotiator = OkHttpProtocolNegotiator.createNegotiator(cl);
+
+    assertEquals(OkHttpProtocolNegotiator.AndroidNegotiator.class, negotiator.getClass());
+  }
+
+  @Test
+  public void createNegotiator_isAndroidLegacy() {
+    ClassLoader cl = new ClassLoader(this.getClass().getClassLoader()) {
+      @Override
+      protected Class<?> findClass(String name) throws ClassNotFoundException {
+        // Just don't throw.
+        if ("org.apache.harmony.xnet.provider.jsse.OpenSSLSocketImpl".equals(name)) {
+          return null;
+        }
+        return super.findClass(name);
+      }
+    };
+
+    OkHttpProtocolNegotiator negotiator = OkHttpProtocolNegotiator.createNegotiator(cl);
+
+    assertEquals(OkHttpProtocolNegotiator.AndroidNegotiator.class, negotiator.getClass());
+  }
+
+  @Test
+  public void createNegotiator_notAndroid() {
+    ClassLoader cl = new ClassLoader(this.getClass().getClassLoader()) {
```

Comment:
Maybe not, but explicit is better than implicit.  This makes it clear that if not on android it should fail.